### PR TITLE
Update ValueGeneratorBase.cs

### PR DESCRIPTION
**Pull request by WISNI4**
string = datetime format error fix for c#

### CODE REVIEW

**Comment by WISNI4** [comment]
string = datetime type error fix for c#

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs [hunk 1]
```diff
@@ -15,7 +17,17 @@ namespace NJsonSchema.CodeGeneration
     public abstract class ValueGeneratorBase
     {
         private readonly CodeGeneratorSettingsBase _settings;
-
+        private readonly List<string> _fomatNotCompatibileWithString = new List<string>() {
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: Typo

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs [hunk 2]
```diff
@@ -15,7 +17,17 @@ namespace NJsonSchema.CodeGeneration
...
+            JsonFormatStrings.DateTime,
+            JsonFormatStrings.Time,
+            JsonFormatStrings.TimeSpan,
+            JsonFormatStrings.Uri,
+            JsonFormatStrings.Guid,
+            JsonFormatStrings.Uuid,
+            JsonFormatStrings.Base64,
+            JsonFormatStrings.Byte,
+        }; 
         /// <summary>Initializes a new instance of the <see cref="ValueGeneratorBase" /> class.</summary>
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: Missing line break

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs [hunk 3]
```diff
@@ -15,7 +17,17 @@ namespace NJsonSchema.CodeGeneration
...
     {
         private readonly CodeGeneratorSettingsBase _settings;
-
+        private readonly List<string> _fomatNotCompatibileWithString = new List<string>() {
+            JsonFormatStrings.Date,
+            JsonFormatStrings.DateTime,
+            JsonFormatStrings.Time,
+            JsonFormatStrings.TimeSpan,
+            JsonFormatStrings.Uri,
+            JsonFormatStrings.Guid,
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: The exclusion of uri guid and uuid should only be applied to the csharp value generator and not to the typescript one...

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs [hunk 4]
```diff
@@ -51,6 +64,15 @@ public virtual string GetDefaultValue(JsonSchema4 schema, bool allowsNull, strin
             return null;
         }
 
+        private string GetStringValue(JsonObjectType type, object value, string format)
+        {
+            if(_fomatNotCompatibileWithString.Contains(format))
+                return null;
+            else
+                return "\"" + ConversionUtilities.ConvertToStringLiteral(value.ToString()) + "\"";
+            //TODO : eg. DateTime.Prase("\"" + ConversionUtilities.ConvertToStringLiteral(value.ToString()) + "\""); for c#
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: Please create a new issue for this todo and remove the todo

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs (hunk 2 above)
Comment (WISNI4) [reply src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: Done

File: src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs (hunk 4 above)
Comment (WISNI4) [reply src/NJsonSchema.CodeGeneration/ValueGeneratorBase.cs]: Done

**Comment by WISNI4** [comment]
i've resolved issues, check please

**Pull request merged**